src/csharp/Record/Exceptions/AzureKinectAddAttachmentException.cs

```diff
@@ -0,0 +1,101 @@
+        /// </summary>
+        /// <param name="message">The message that describes the error.</param>
+        public AzureKinectAddAttachmentException(string message)
+            : base(message)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureKinectAddAttachmentException"/> class
+        /// with a specified error message and a reference to the inner exception that is the
+        /// cause of this exception.
+        /// </summary>
+        /// <param name="message">
+        /// The error message that explains the reason for the exception.
+        /// </param>
+        /// <param name="innerException">
+        /// The exception that is the cause of the current exception, or a null reference
+        /// (Nothing in Visual Basic) if no inner exception is specified.
+        /// </param>
+        public AzureKinectAddAttachmentException(string message, Exception innerException)
+            : base(message, innerException)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureKinectAddAttachmentException"/> class
+        /// with serialized data.
+        /// </summary>
+        /// <param name="info">
+        /// The <see cref="SerializationInfo"/> that holds the serialized object data about the
+        /// exception being thrown.</param>
+        /// <param name="context">
+        /// The <see cref="StreamingContext"/>System.Runtime.Serialization.StreamingContext that
+        /// contains contextual information about the source or destination.
+        /// </param>
+        protected AzureKinectAddAttachmentException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureKinectAddAttachmentException"/> class
+        /// with a specified error message.
+        /// </summary>
+        /// <param name="message">The message that describes the error.</param>
+        /// <param name="logMessages">
+        /// The log messages that happened during the function call that generated this error.
+        /// </param>
+        protected AzureKinectAddAttachmentException(string message, ICollection<LogMessage> logMessages)
+            : base(message, logMessages)
+        {
+        }
+
+        /// <summary>
+        /// Throws an <see cref="AzureKinectAddAttachmentException"/> if the result of the function
+        /// is not a success.
+        /// </summary>
+        /// <param name="function">The native function to call.</param>
+        /// <typeparam name="T">The type of result to expect from the function call.</typeparam>
+        internal static void ThrowIfNotSuccess<T>(Func<T> function)
+            where T : System.Enum
+        {
+            using (LoggingTracer tracer = new LoggingTracer(LogLevel.Warning, Logger.LogProvider, RecordLogger.LogProvider))
```

Comment:
Should we do something to make it easier to update the LogLevel?  Incase for debug you need a higher logging level?

Reply:
You can always wrap in your own LoggerTracer and capture the higher verbosity from another level, so I'd just stick with that.